### PR TITLE
Use exports.__esModule = true in package sub-path exports so Rollup recognizes ESM

### PR DESCRIPTION
**Pull request by jayphelps**
This PR is more like an issue ticket that happens to provide one possible solution, just because it was trivial to do.

## The Problem

tl;dr if already familiar with `__esModule` is that Next.js's intermediate CJS files prevent Rollup from recognizing default exports correctly.

---

If you use Rollup and try and import the default export from one of the `next/*` files e.g. `import Link from "next/link"` it will incorrectly resolve to the full `exports` object itself that contains the `exports.default` on it.

The cause is, as these things are, a bit involved, but the gist is that [`@rollup/commonjs`](https://github.com/rollup/plugins/tree/master/packages/commonjs) is used to take packages and files that have CommonJS exports and automatically attempt to convert them to the equivalent ES Module export. ESM's behavior of a "default export" is syntactic sugar for a named property called `default`,  but the CommonJS conventions don't use that pattern, but rather just exporting the value itself as the entire export `modules.exports = value`.

It's easy enough to just say the interop is that `module.export` in CJS files should be treated as if it were the `export default` of an ESM. Trouble is, there are desirable reasons for ESM compilers to emit CJS that does not use `module.exports = value` but instead uses `exports.default = value` e.g. if the default export is a function, it would have to assign the other named exports to it, which is a non-starter.

To solve the inconsistency a convention the community has settled on is that if a CommonJS file has an `exports.__esModule = true` that it was originally written using ESM but was compiled down to `exports.default` property, so tooling knows to treat it as such.

Next.js internally uses TypeScript for most things, however has [hand written CJS files](https://github.com/vercel/next.js/blob/b403e9ec15668c98a802c42bd17cb3f3678535cd/packages/next/link.js#L1) that act as the actual entry points when imported from the published package. e.g. just simply `module.exports = require('./dist/client/link')`. This is a problem with Rollup (and presumably tooling like it) because while [the TypeScript compiled output has `exports.__esModule`](https://unpkg.com/next@10.0.3/dist/client/link.js), Rollup doesn't treat it as such because of this intermediate CJS file.

## The Use Case

Admittedly, this is a weird one for Next.js because "importing next.js packages from Rollup" is probably not something you thought anyone would try to do. 😅 

We at @outsmartly [have a product](https://outsmartly.org/features) that _other developers_ use with their regular Next.js apps. Long story short, it does static analysis and program slicing do to fast, selective re-rendering of only a portion of your code, at the Edge. One part of that magic leans on Rollup to bundle some of the developer's code from their normal Next.js app. Code that uses things like `import Link from 'next/link'` and you can probably see where this is going 😆 that these intermediate files break that.

It took me quite a bit of digging to realize the issue, as seen in my demo of the problem here: https://github.com/jayphelps/rollup-cjs-issue

## The Solution(s)

1. The most obvious solution is to just add `exports.__esModule = true;` to each of these files and call it a day. The pros is that it's simple, requires very little changes (just the ones included in this PR) and is unlikely to break anyone, unless they happen to be also running into this issue and just working around it using `import link from 'next/link'; const { default: Link = link;` which is unlikely (but I bring it up for transparency.) Even if they were, they would probably be thrilled this was fixed.

2. Alternatively, one could change these offending CJS files to be TS files too, thus acquiring the `exports.__esModule` automatically. This was [tried two years ago](https://github.com/timneutkens/next.js/commit/15bb1c5e7963361c9c592923ef28cd819092e49a#diff-299477910394fec52f35b42306ed660762c2967d5b8ef775a6d5876c53fc2747L1) but then was quickly reverted #5815 due to backwards compatibility issues. It is unclear whether these issues have since been resolved.

3. Do nothing. Which is obviously bad for me and @outsmartly, but I can imagine that supporting Rollup is quite low on the list, if it even is at all so I at least want to acknowledge that I realize this is an unusual request.

:shipit:

### CODE REVIEW

File: packages/next/amp.js [hunk 1]
```diff
@@ -1 +1,2 @@
 module.exports = require('./dist/next-server/lib/amp')
+exports.__esModule = true;
```
Comment (jayphelps) [inline packages/next/amp.js:2]: Even though the files being require()d here will have their own `exports.__esModule = true`, so this code is redundant from a runtime perspective, Rollup relies on static analysis, ahead of time, so it needs to be inside these files to be recognized.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| buildDuration | 10.3s | 9.6s | -706ms |
| nodeModulesSize | 82.5 MB | 82.5 MB | ⚠️ +216 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.34 | 2.302 | -0.04 |
| / avg req/sec | 1068.56 | 1086.14 | +17.58 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.252 | 1.232 | -0.02 |
| /error-in-render avg req/sec | 1996.24 | 2028.76 | +32.52 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| 677f882d2ed8..34b9.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-f501b4a..c42e.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-01375..90b6.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-db223d9..dbd7.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 8.27 kB | 8.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| Overall change | 322 B | 322 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 621 B | 621 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| buildDuration | 11.4s | 11s | -307ms |
| nodeModulesSize | 82.5 MB | 82.5 MB | ⚠️ +216 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| 677f882d2ed8..34b9.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-f501b4a..c42e.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-01375..90b6.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-db223d9..dbd7.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 8.27 kB | 8.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| Overall change | 322 B | 322 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.16 MB | 5.16 MB | ✓ |

</details>

</details>
Commit: abff4c2274955b3a9b5510362e22d86ce97a35c1

**Comment by jayphelps** [comment]
Apologies, I did not knowingly request anyone's specific review. 

image

**Comment by timneutkens** [comment]
> Apologies, I did not knowingly request anyone's specific review.

That's automatic on PRs 👍

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| buildDuration | 9.2s | 9.4s | ⚠️ +180ms |
| nodeModulesSize | 82.5 MB | 82.5 MB | ⚠️ +260 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.278 | 2.295 | ⚠️ +0.02 |
| / avg req/sec | 1097.61 | 1089.32 | ⚠️ -8.29 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.197 | 1.165 | -0.03 |
| /error-in-render avg req/sec | 2088.2 | 2146.42 | +58.22 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| 677f882d2ed8..34b9.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-f501b4a..c42e.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-01375..90b6.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-db223d9..dbd7.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 8.27 kB | 8.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| Overall change | 322 B | 322 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 621 B | 621 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| buildDuration | 11s | 10.9s | -123ms |
| nodeModulesSize | 82.5 MB | 82.5 MB | ⚠️ +260 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| 677f882d2ed8..34b9.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-f501b4a..c42e.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-01375..90b6.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-db223d9..dbd7.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 8.27 kB | 8.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| Overall change | 322 B | 322 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.16 MB | 5.16 MB | ✓ |

</details>

</details>
Commit: 55647396f12a034c4db00931ea71c5d91ffdb034

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| buildDuration | 9.9s | 9.9s | ⚠️ +22ms |
| nodeModulesSize | 82.5 MB | 82.5 MB | ⚠️ +260 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.425 | 2.42 | 0 |
| / avg req/sec | 1030.78 | 1033.26 | +2.48 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.479 | 1.486 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 1690.23 | 1682.3 | ⚠️ -7.93 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| 677f882d2ed8..34b9.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-f501b4a..c42e.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-01375..90b6.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-db223d9..dbd7.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 8.27 kB | 8.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| Overall change | 322 B | 322 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 621 B | 621 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| buildDuration | 12.3s | 12.2s | -153ms |
| nodeModulesSize | 82.5 MB | 82.5 MB | ⚠️ +260 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| 677f882d2ed8..34b9.js gzip | 12.8 kB | 12.8 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-f501b4a..c42e.js gzip | 6.56 kB | 6.56 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59 kB | 59 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-01375..90b6.js gzip | 3.69 kB | 3.69 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-db223d9..dbd7.js gzip | 1.61 kB | 1.61 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 8.27 kB | 8.27 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| Overall change | 322 B | 322 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | jayphelps/next.js rollup | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.16 MB | 5.16 MB | ✓ |

</details>

</details>
Commit: e09928064223ba85381696f6278c2d4ad11db5bc

**Comment by ijjk** [comment]
## Failing test suites 
Commit: e09928064223ba85381696f6278c2d4ad11db5bc

**test/integration/production-build-dir/test/index.test.js** 
- Production Custom Build Directory > With basic usage > should render the page

<details>
<summary>Expand output</summary>

  ● Production Custom Build Directory › With basic usage › should render the page

    expect(received).toBe(expected) // Object.is equality

    Expected: ""
    Received: "The postprocess middleware \"Inline-Fonts\" took 14ms(14, 0) to complete. This is longer than the 10 limit.
    "

      20 |         stderr: true,
      21 |       })
    > 22 |       expect(result.stderr).toBe('')
         |                             ^
      23 | 
      24 |       const app = nextServer({
      25 |         dir: join(__dirname, '../build'),

      at Object.<anonymous> (integration/production-build-dir/test/index.test.js:22:29)

</details>
